plugins/module_utils/rds.py

```diff
@@ -542,6 +589,44 @@ def arg_spec_to_rds_params(options_dict: Dict[str, Any]) -> Dict[str, Any]:
     return camel_options
 
 
+def format_rds_client_method_parameters(
+    client, module: AnsibleAWSModule, parameters: Dict[str, Any], method_name: str, format_tags: bool
+) -> Dict[str, Any]:
+    """
+    Returns a dict of parameters validated and formatted for the provided boto3 client method.
+
+    Performs the following parameters checks and updates:
+        - Converts parameters supplied as snake_cased module options to CamelCase
+        - Ensures that all required parameters for the provided method are present
+        - Ensures that only parameters allowed for the provided method are present, removing any that are not relevant
+        - Removes parameters with None values
+        - If format_tags is True, converts "Tags" param from an Ansible dict to boto3 list of dicts
+
+        Parameters:
+            client: boto3 rds client
+            module: AnsibleAWSModule
+            parameters (dict): Parameter options as provided to module
+            method_name (str): boto3 client method for which to validate parameters
+            format_tags (bool): Whether to convert tags from an Ansible dict to boto3 list of dicts
+
+        Returns:
+            Dict of client parameters formatted for the provided method
+
+        Raises:
+            Fails the module if any parameters required by the provided method are not provided in module options
+    """
+    required_options = get_boto3_client_method_parameters(client, method_name, required=True)
+    if any(parameters.get(k) is None for k in required_options):
+        method_description = get_rds_method_attribute(method_name, module).operation_description
+        module.fail_json(msg=f"To {method_description} requires the parameters: {required_options}")
+    options = get_boto3_client_method_parameters(client, method_name)
+    parameters = dict((k, v) for k, v in parameters.items() if k in options and v is not None)
```

Comment:
what if we raise an error for unsupported value into parameters?
```
if any(k not in options for k, v in parameters.items()):
   module.fail_json(msg=f"unsupported option {k} for method name {method_name}")
```

Reply:
That would change the user experience, failing the module in cases where it does not currently; right now it just ignores any unsupported parameters and uses the ones that are relevant. I think that's probably a good change but I'd rather not make it in a refactor PR.